packages/database/src/models/SurveyResponse.ts

```diff
@@ -388,12 +388,25 @@ export class SurveyResponse extends Model {
       if (!dataElement) {
         throw new Error(`no data element for question: ${dataElementId}`);
       }
-      const body = getStringValue(dataElement.type, value);
+      let body = getStringValue(dataElement.type, value);
       // Don't create null answers
       if (body === null) {
         continue;
       }
 
+      if (dataElement.type === PROGRAM_DATA_ELEMENT_TYPES.PHOTO) {
```

Comment:
I think this block suits better inside `getStringValue`. That way:
- We aren't running `getStringValue` with other code unnecessarily when it is a photo
- All data element types are handled in the same util function
- Any other checks like the `body === null` above get run on photo answers like all others, no risk of missing it from something common

Reply:
You've thought about it a lot harder than I have, thanks for all your reasoning and happy to go with whatever you choose!